sw/device/lib/crypto/drivers/entropy.c

```diff
@@ -230,6 +230,16 @@ OT_WARN_UNUSED_RESULT
 static status_t csrng_send_app_cmd(uint32_t reg_address,
                                    entropy_csrng_cmd_t cmd,
                                    bool check_completion) {
+  enum {
+    // This is to maintain full compliance with NIST SP 800-90A, which requires
+    // the max generate output to be constrained to gen < 2 ^ 12 bits or 0x800
+    // 128-bit blocks.
+    kMaxGenerateSizeIn128BitBlocks = 0x800,
+  };
+  if (cmd.generate_len > kMaxGenerateSizeIn128BitBlocks) {
+    return OUT_OF_RANGE();
```

Comment:
nit: sorry for the post-merge comment, but it might be good to use a cryptolib status macro here so it's clear which top-level error code it will become:
```suggestion
return OTCRYPTO_BAD_ARGS;
```

As written, this will get translated into an "internal error" in the top-level API, which is the default for most error codes that aren't specifically mapped to one of the top-level status codes.
https://github.com/lowRISC/opentitan/blob/39f9a9e57286f0d2a72c624bb19b9bd3a4c3bdf0/sw/device/lib/crypto/impl/status.c#L43-L44